applications/test_channel/src/lib.rs

```diff
@@ -404,3 +434,63 @@ fn print_usage(opts: Options) {
 
 const USAGE: &'static str = "Usage: test_channel OPTION ARG
 Provides a selection of different tests for channel-based communication.";
+
+/// Create one sender and one receiver. Drop the receiver. Channel should be marked as disconnected. 
+fn drop_last_sender_oneshot<T:Send>() -> Result<(), &'static str> {
+    let (sender, receiver) = async_channel::new_channel::<T>(2);
+    drop(sender);
+    
+    if receiver.channel.channel_status.as_ptr() != &mut async_channel::ChannelStatus::SenderDisconnected {
+        warn!("WARNING: failed to change ChannelStatus to SenderDisconnected for original receiver");
+    }
+    
+    // not sure how to test whether receivers were notified
+    Ok(())
+}
+
+/// Create two receivers and one sender. Drop the sender. Channel should be marked as disconnected. 
+fn drop_last_sender_multiple<T:Send>() -> Result<(), &'static str> {
+    let (sender, receiver) = async_channel::new_channel(2);
+    let receiver2: async_channel::Receiver<T> = receiver.clone();
+    drop(sender);
+    
+    if receiver.channel.channel_status.as_ptr() != &mut async_channel::ChannelStatus::SenderDisconnected {
+        warn!("WARNING: failed to change ChannelStatus to SenderDisconnected for original receiver");
```

Comment:
warning is good, but asserting or throwing an error would be even better. It is a test, after all, so failures should be loud.